installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -386,8 +410,18 @@ public static ActionResult StateDeploy(Session session)
                         record.FormatString = String.Format("{0} failed with error:\n{1}", seq.Description, errorOutput);
 
                         MessageResult msgRes = session.Message(InstallMessage.Error | (InstallMessage)MessageBoxButtons.OK, record);
+                        if (!artifactsFetched)
+                        {
+                            TrackerSingleton.Instance.TrackEventSynchronously(session, "stage", "artifacts", "failure");
```

Comment:
Artifacts in this case refers to the whole deployment job. So you can call this for the entirety of the deploy process.